vllm/model_executor/layers/quantization/utils/fp8_utils.py

```diff
@@ -240,87 +250,87 @@ def _run_deepgemm(
         self,
         input_2d: torch.Tensor,
         weight: torch.Tensor,
-        block_size: list[int],
         weight_scale: torch.Tensor,
     ) -> torch.Tensor:
         # ensure DeepGEMM-backed custom op is registered before use
         import vllm.model_executor.layers.quantization.deepgemm  # noqa: F401
 
-        q_input, x_scale = per_token_group_quant_fp8(
-            input_2d,
-            block_size[1],
-            column_major_scales=True,
-            use_ue8m0=self.ue8m0_deepgemm_supported,
-        )
+        assert self.deepgemm_input_quant_op is not None
+        q_input, x_scale = self.deepgemm_input_quant_op.forward_cuda(input_2d)
```

Comment:
What would it take to be able to run `forward_native` here? Adding `use_e8m0` support to the native path? If yes can we try to do that?

Reply:
`use_e8m0` should be true only when we use deepgemm, but it seems to only affect one line in `per_token_group_quant_fp8`:
```y_s = tl.math.exp2(tl.ceil(tl.log2(scale_raw))) if use_ue8m0 else scale_raw```
So probably
```        scales = (absmax / _FP8_MAX).clamp(min=_FP8_MIN_SCALING_FACTOR)```
should be changed to
```
        scale_raw = absmax / _FP8_MAX
        if self.use_ue8m0:
            scale_raw = torch.exp2(torch.ceil(torch.log2(scale_raw)))
        scales = (scale_raw).clamp(min=_FP8_MIN_SCALING_FACTOR)
```
?

Reply:
Ok, I added it + expanded unit test in `tests/kernels/quantization/test_fp8_quant_group.py` to capture it